File: eCinema.Model/Dvorana.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCinema.Model
{
    public class Dvorana
    {
        public int DvoranaId { get; set; }
        public string Naziv { get; set; }
        public int BrSjedista { get; set; }
        public int BrRedova { get; set; }
        public int BrSjedistaPoRedu { get; set; }

        public int? CinemaId { get; set; }
        public virtual Cinema? Cinema { get; set; }

        public override string ToString()
        {
            return Naziv;
        }
    }
}

File: ecinema_mobile/lib/models/dvorana.dart
import 'package:ecinema_mobile/models/cinema.dart';
import 'package:json_annotation/json_annotation.dart';

part 'dvorana.g.dart';

@JsonSerializable()
class Dvorana {
  int dvoranaId;
  String naziv;
  int brSjedista;
  int brRedova;
  int brSjedistaPoRedu;
  int? cinemaId;
  Cinema? cinema;

  Dvorana(
      {required this.dvoranaId,
      required this.naziv,
      required this.brSjedista,
      required this.brRedova,
      required this.brSjedistaPoRedu,
      this.cinemaId,
      this.cinema});

  factory Dvorana.fromJson(Map<String, dynamic> json) =>
      _$DvoranaFromJson(json);
  Map<String, dynamic> toJson() => _$DvoranaToJson(this);
}
